pkg/routers/api.go

```diff
@@ -144,19 +144,19 @@ func WithAPIRouter(opt *APIOptions, r chi.Router) {
 
 	r.Route("/v1", func(r chi.Router) {
 		r.Use(opt.corsHandler, opt.sdkMiddleware)
-		r.With(getConfigTimer, opt.oAuthMiddleware).Get("/config", opt.configHandler)
-		r.With(getDatafileTimer, opt.oAuthMiddleware).Get("/datafile", opt.datafileHandler)
-		r.With(activateTimer, opt.oAuthMiddleware, contentTypeMiddleware).Post("/activate", opt.activateHandler)
-		r.With(decideTimer, opt.oAuthMiddleware, contentTypeMiddleware).Post("/decide", opt.decideHandler)
-		r.With(trackTimer, opt.oAuthMiddleware, contentTypeMiddleware).Post("/track", opt.trackHandler)
-		r.With(overrideTimer, opt.oAuthMiddleware, contentTypeMiddleware).Post("/override", opt.overrideHandler)
-		r.With(lookupTimer, opt.oAuthMiddleware, contentTypeMiddleware).Post("/lookup", opt.lookupHandler)
-		r.With(saveTimer, opt.oAuthMiddleware, contentTypeMiddleware).Post("/save", opt.saveHandler)
-		r.With(sendOdpEventTimer, opt.oAuthMiddleware, contentTypeMiddleware).Post("/send-odp-event", opt.sendOdpEventHandler)
-		r.With(opt.oAuthMiddleware).Get("/notifications/event-stream", opt.nStreamHandler)
+		r.With(getConfigTimer, opt.oAuthMiddleware, middleware.AddTracing("configHandler", "OptimizelyConfig")).Get("/config", opt.configHandler)
+		r.With(getDatafileTimer, opt.oAuthMiddleware, middleware.AddTracing("datafileHandler", "OptimizelyDatafile")).Get("/datafile", opt.datafileHandler)
+		r.With(activateTimer, opt.oAuthMiddleware, contentTypeMiddleware, middleware.AddTracing("activateHandler", "Activate")).Post("/activate", opt.activateHandler)
```

Comment:
Wondering why we do tracing here and also in ActivateHandler redundantly?

Reply:
Actually from Activate Handler, we are calling some more important method. We are covering those too. It'll provide trace data like a chain.

Reply:
I now see the point.  Not sure about the value of adding tracing for the sub levels since they won't care or may be able to differenticate from attributes?